thoth/glyph/lib.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import os
+from os import path
+from thoth.common import init_logging
+from fasttext import load_model
+
+import pandas as pd
+from pygit2 import Repository
+from pygit2 import GIT_SORT_TOPOLOGICAL
+
+import time
+import datetime
+import sys
+
+init_logging()
+_LOGGER = logging.getLogger("glyph")
+DEFAULT_MODEL_PATH = path.join(path.dirname(__file__), 'data/model_commits_v2_quant.bin')
+
+
+def classify_by_date(path, start, end, output, model):
+    start_time = 0
+    end_time = sys.maxsize
+
+    if start is not None:
+        start_time = int(time.mktime(datetime.datetime.strptime(start, "%Y-%m-%d").timetuple()))
+
+    if end is not None:
+        end_time = int(time.mktime(datetime.datetime.strptime(end, "%Y-%m-%d").timetuple()))
+
+    repo_path = os.path.join(path, ".git")
+    if os.path.exists(repo_path):
+        repo = Repository(repo_path)
+    else:
+        _LOGGER.error("Git repository not found")
+        return
```

Comment:
Usually it's a better idea to define exceptions so that library user can act on errors encountered.

Reply:
Should I define a my own custom exceptions?

Reply:
Yes, that's a good idea. A hierarchy of thoth-glyph specific exceptions.

Reply:
Added a couple of custom exceptions:
https://github.com/thoth-station/glyph/pull/13/files#diff-f3b0cb6c056c89006d09c850c125774a